garrysmod/lua/includes/extensions/vector.lua

```diff
@@ -3,10 +3,247 @@ local meta = FindMetaTable( "Vector" )
+function meta:GetRotated(...)
+	local v = Vector(self)
+	v:Rotate(...)
+	return v
+end
+
+--[[---------------------------------------------------------
+Returns a copy of a multiplied vector
+-----------------------------------------------------------]]
+function meta:GetMul(...)
+	local v = Vector(self)
+	v:Mul(...)
+	return v
+end
+
+--[[---------------------------------------------------------
+Returns a copy of an added vector
+-----------------------------------------------------------]]
+function meta:GetAdd(...)
+	local v = Vector(self)
+	v:Add(...)
+	return v
+end
+
+--[[---------------------------------------------------------
+Adds a unpacked x, y and z to a vector
+-----------------------------------------------------------]]
+function meta:AddEx(x, y, z)
+	self[1] = self[1] + x
+	self[2] = self[2] + y
+	self[3] = self[3] + z
+	return self
+end
+
+--[[---------------------------------------------------------
+Adds a unpacked x, y and z to a copy vector
+-----------------------------------------------------------]]
+function meta:GetAddEx(...)
+	local v = Vector(self)
+	v:AddEx(...)
+	return self
+end
+
+--[[---------------------------------------------------------
+Returns a copy of a subracted vector
+-----------------------------------------------------------]]
+function meta:GetSub(...)
+	local v = Vector(self)
+	v:Sub(...)
+	return v
+end
+
+--[[---------------------------------------------------------
+Subtracts a unpacked x, y and z from a vector
+-----------------------------------------------------------]]
+function meta:SubEx(x, y, z)
+	self[1] = self[1] - x
+	self[2] = self[2] - y
+	self[3] = self[3] - z
+	return self
+end
+
+--[[---------------------------------------------------------
+Adds a unpacked x, y and z to a copy vector
+-----------------------------------------------------------]]
+function meta:GetSubEx(...)
+	local v = Vector(self)
+	v:SubEx(...)
+	return self
+end
+
+--[[---------------------------------------------------------
+Returns a copy of a divided vector
+-----------------------------------------------------------]]
+function meta:GetDiv(...)
+	local v = Vector(self)
+	v:Div(...)
+	return v
+end
+
+--[[---------------------------------------------------------
+Convets the vector to bisector of two vectors
+-----------------------------------------------------------]]
+function meta:Bisect(vec)
+	local ns, nv = self:Length(), vec:Length()
+	self:Mul(nv)
+	self:Add(vec:GetMul(ns))
+	return self
+end
+
+--[[---------------------------------------------------------
+Returns a copy of bisector of two vectors
+-----------------------------------------------------------]]
+function meta:GetBisected(...)
+	local v = Vector(self)
+	v:Bisect(...)
+	return v
+end
+
+--[[---------------------------------------------------------
+Convets the vector to nagated version of itself
+-----------------------------------------------------------]]
+function meta:Negate(bx, by, bz)

```

Comment:
I think these arguments are pointless. Negate should negate the entire vector and not just parts - that can be done inline in code if necessary instead of inducing a bunch of type checks.

Reply:
Ok. Thanks !